File: 2025/02/prog_12909_올바른괄호.py
def solution(s):
    answer = True

    stack = []
    
    for ss in s:
        if ss == '(':
            stack.append(ss)
        else:
            # stack이 비어있다면 실패 
            if not stack:
                return False
            else:
                stack.pop()
                
    # stack이 차있다면 실패 
    if stack:
        return False
    
    return True

File: 2025/07/prog_12909_올바른괄호.js
function solution(s){
    let stack = []
    
    for (let i=0; i < s.length; i++) {
        const ss = s[i]
        
        if (ss === '(') {
            stack.push(true)
        } else {
            if (stack.length === 0) {
                return false
            }
            
            stack.pop()
        }
    }
    
    if (stack.length !== 0) {
        return false
    }

    return true;
}